pkg/cmd/server/start/kube_connection_args.go

```diff
@@ -24,19 +24,29 @@ type KubeConnectionArgs struct {
 	ClientConfigLoadingRules clientcmd.ClientConfigLoadingRules
 }
 
+// BindKubeConnectionArgs binds values to the given arguments by using flags
 func BindKubeConnectionArgs(args *KubeConnectionArgs, flags *pflag.FlagSet, prefix string) {
 	// TODO remove entirely
 	flags.Var(&args.KubernetesAddr, prefix+"kubernetes", "removed in favor of --"+prefix+"kubeconfig")
 	flags.StringVar(&args.ClientConfigLoadingRules.ExplicitPath, prefix+"kubeconfig", "", "Path to the kubeconfig file to use for requests to the Kubernetes API.")
 }
 
+// NewDefaultKubeConnectionArgs returns a new set of default connection
+// arguments for Kubernetes
 func NewDefaultKubeConnectionArgs() *KubeConnectionArgs {
-	config := &KubeConnectionArgs{}
+	return &KubeConnectionArgs{
+		KubernetesAddr: flagtypes.Addr{
+			Value: "localhost:8443", 
+			DefaultScheme: "https", 
+			DefaultPort: 8443, 
+			AllowPrefix: true,
+		}.Default(),
 
-	config.KubernetesAddr = flagtypes.Addr{Value: "localhost:8443", DefaultScheme: "https", DefaultPort: 8443, AllowPrefix: true}.Default()
```

Comment:
This changes the behavior of this call in a breaking way - &config is no longer defined.